scripts/build-js/build-js.js

```diff
@@ -0,0 +1,59 @@
+const fs = require("fs");
+const fsPromises = require("fs").promises;
+const jsBase = fs.readFileSync(__dirname + "/heroicons-js-base.js", "utf8");
+
+const createLookupTable = (keys, values) => {
+  return keys.reduce((acc, curr, index) => {
+    const keyName = curr.replace(".svg", "");
+    return {
+      ...acc,
+      [keyName]: values[index],
+    };
+  }, {});
+};
+
+const OUTLINE_ICONS_PATH = `./outline`;
+const SOLID_ICONS_PATH = `./solid`;
+
+const getIconCodeArray = async (path, fileNames) =>
+  await Promise.all(
+      fileNames.map((icon) => fsPromises.readFile(`${path}/${icon}`, "utf8"))
+  );
+
+const createLookupTables = async () => {
+  const solidIconsArray = await fsPromises.readdir("./solid");
+  const outlineIconsArray = await fsPromises.readdir("./outline");
+
+  const solidIcons = await getIconCodeArray(
+    SOLID_ICONS_PATH,
+    solidIconsArray
+  );
+  const outlineIcons = await getIconCodeArray(
+    OUTLINE_ICONS_PATH,
+    outlineIconsArray
+  );
+
+  const solidLookupTable = createLookupTable(solidIconsArray, solidIcons);
+  const outlineLookupTable = createLookupTable(
+    outlineIconsArray,
+    outlineIcons
+  );
+
+  return {
+    solidLookupTable,
+    outlineLookupTable,
+  };
+};
+
+createJsFile = async () => {
+  const lookupTables = await createLookupTables();
+  const solid = JSON.stringify(lookupTables.solidLookupTable);
+  const outline = JSON.stringify(lookupTables.outlineLookupTable);
+  const filePrefix = `const solid = ${solid};\n const outline = ${outline};\n`;
```

Comment:
Your script seem not to be a module, because of the missing export. This would mean your are introducing two global variables with extreme generic names `solid` and `outline`. This can crash someones code, if there are such global variables too.

Reply:
I was a little confused about what you meant here. The `solid` and `outline` variables are within the function so I'm not expecting them to be global. I did notice, however, that I missed the `const` declaration so I was declaring the whole function globally (perhaps this is what you meant) so I've updated this now.